universe/ScriptingContext.h

```diff
@@ -9,6 +9,22 @@
 
 class UniverseObject;
 
+struct BackgroundContext {
+public:
+    BackgroundContext()
+    {}
+
+    BackgroundContext(int bout_, Universe::EmpireObjectVisibilityMap& empire_object_visibility_) :
+        bout(bout_),
+        empire_object_visibility(empire_object_visibility_)
+    {}
+
+    int bout;
+    Universe::EmpireObjectVisibilityMap empire_object_visibility;       ///< indexed by empire id and object id, the visibility level the empire has of each object.  may be increas
+
+};
+static const BackgroundContext theEmptyBackground = BackgroundContext();
```

Comment:
Make this a vis map static class variable `s_empty_vis_map`. Don't need to assign it any value; the default empty state should work for the dummy / default thing to bind the reference to.

Reply:
I could not make the static map work. After lots of trials i have a working version where i pass a reference to the CombatInfo (where i added a int bout member). 

Could not make a static const class member work as dummy either it got multiple defined, not initialised or the symbol was hidden. Works a global though.